### PR TITLE
Change the docker compose commands to be functions

### PR DESCRIPTION
**Pull request by kmcnellis**
### Description
Avoids using variables for commands, removed use of exec.

### Checklist
- [ ] Ran all the scripts

### CODE REVIEW

File: bin/lib/docker.sh [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
-  -f docker-compose.yml \
-  -f browser-test/browser-test-compose.yml \
-  -f browser-test/browser-test-compose.dev.yml"
+function docker::compose_browser_test_dev() {
+  docker compose \
+    -f docker-compose.yml \
+    -f browser-test/browser-test-compose.yml \
+    -f browser-test/browser-test-compose.dev.yml \
+    "$@"
+}
```
Comment (bion) [inline bin/lib/docker.sh:35]: suggestion (blocking): document these functions following [the style guide](https://google.github.io/styleguide/shellguide.html#s4.2-function-comments)

File: bin/run-browser-test-env [hunk 2]
```diff
@@ -75,46 +75,59 @@ EOF
...
+#######################################
+# Chooses the dev or ci versions of broswertest
+# Arguments:
+#   1: if true, use developer mode
+#   2: the rest of the arguments for compose
+#######################################
+function compose_browser_test() {
+  local use_dev="$1"
+  shift 1
+  if [ "$use_dev" = true ]; then
```
Comment (bion) [inline bin/run-browser-test-env:87]: suggestion (blocking): double brackets to conform to style guide

File: bin/run-browser-test-env [hunk 3]
```diff
@@ -75,46 +75,59 @@ EOF
...
+    docker::compose_browser_test "$@"
+  fi
+}
+
 set_args "$@"
 
 bin/pull-image
 
-base_compose_command="${DOCKER_COMPOSE_BROWSER_TEST}"
+use_dev=false
```
Comment (bion) [inline bin/run-browser-test-env:100]: suggestion (blocking): `truth::declare_var_false use_dev`

File: bin/run-browser-test-env [hunk 4]
```diff
@@ -75,46 +75,59 @@ EOF
...
 if truth::is_disabled continuous_integration; then
   # If not running in CI, load the dev addition to the browser test compose
   # this mounts the local SBT cache directories for faster subsequent builds.
-  base_compose_command="${DOCKER_COMPOSE_BROWSER_TEST_DEV}"
+  use_dev=true
```
Comment (bion) [inline bin/run-browser-test-env:104]: suggestion (blocking): `truth:enable use_dev`

File: bin/run-browser-test-env (hunk 2 above)
Comment (bion) [reply bin/run-browser-test-env:87]: ah actually can use the `truth::` functions: `truth::is_enabled use_dev` after making other suggested changes

**Review by bion — CHANGES_REQUESTED** [review]
Good improvement!

**Review by clouser2 — APPROVED** [review]
I definitely agree that this is a large readability win. Thanks for improving this!

**Review by bion — APPROVED** [review]

File: bin/lib/docker.sh (hunk 1 above)
Comment (kmcnellis) [reply bin/lib/docker.sh:65]: done

**Pull request merged**